### PR TITLE
Set selected Workspace

### PR DESCRIPTION
**Pull request by GarryFirst-cripto**
Set selected Workspace on user fetch/login/sign up

### CODE REVIEW

File: frontend/src/common/models/user/IUser.ts [hunk 1]
```diff
@@ -6,4 +8,5 @@ export interface IUser {
   imageUrl?: string;
   title?: string;
   status?: string;
+  workspaces: IWorkspaceResponse[];
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/user/IUser.ts:11]: you have to create separate interface like in backend

File: frontend/src/common/models/workspace/IWorkspaceResponse.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface IWorkspaceResponse {
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/workspace/IWorkspaceResponse.ts:1]:
```suggestion
export interface IWorkspace {
```

File: frontend/src/scenes/Workspace/reducers/reducer.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-import { addWorkspaceRoutine, selectChatRoutine } from '../routines/routines';
+import { addWorkspaceRoutine, selectChatRoutine, selectWorkspaceRoutine } from '../routines/routines';
 import { IWorkspace } from 'common/models/workspace/IWorkspace';
 
 export interface IWorkspaceState {
   workspace: IWorkspace;
   loading: boolean;
   error: any;
   selectedChat: any;
+  selectedWorkspace: any;
```
Comment (AlikRakhmonov) [inline frontend/src/scenes/Workspace/reducers/reducer.ts:10]: no any

Comment (GarryFirst-cripto) [reply frontend/src/scenes/Workspace/reducers/reducer.ts:10]: Done

File: frontend/src/common/models/user/IUser.ts (hunk 1 above)
Comment (GarryFirst-cripto) [reply frontend/src/common/models/user/IUser.ts:11]: Done

File: frontend/src/common/models/workspace/IWorkspaceResponse.ts (hunk 2 above)
Comment (GarryFirst-cripto) [reply frontend/src/common/models/workspace/IWorkspaceResponse.ts:1]: Done

File: frontend/src/scenes/Workspace/containers/WorkspaceToolbar/index.tsx [hunk 4]
```diff
@@ -3,21 +3,24 @@ import { connect } from 'react-redux';
...
 import WorkspaceItem from './components/WorkspaceItem';
 
+export interface IWorkspaceString {
+  [key: string]: string;
+}
+
 interface IProps {
   fetchWorkspaces: Function;
-  workspaces: IWorkspace[];
+  workspaces: IWorkspaceString[];
```
Comment (AlikRakhmonov) [inline frontend/src/scenes/Workspace/containers/WorkspaceToolbar/index.tsx:18]: incorrect, it should be IWorkspace

File: frontend/src/scenes/Workspace/reducers/reducer.ts [hunk 5]
```diff
@@ -1,21 +1,26 @@
...
+import { IChat } from 'common/models/workstate/chat';
 
 export interface IWorkspaceState {
   workspace: IWorkspace;
   loading: boolean;
-  error: any;
-  selectedChat: any;
-  channels: Array<any>;
-  directMessages: Array<any>;
+  error: string;
```
Comment (AlikRakhmonov) [inline frontend/src/scenes/Workspace/reducers/reducer.ts:9]: incorrect, it should be custom error model

File: frontend/src/scenes/Workspace/reducers/reducer.ts [hunk 6]
```diff
@@ -1,21 +1,26 @@
...
+  selectedWorkspace: IChat;
+  channels: Array<IChat>;
+  directMessages: Array<IChat>;
 }
 
+const emptyChat = { id: '', name: '', isPrivate: false };
+
 const initialState: IWorkspaceState = {
-  workspace: {},
+  workspace: { id: '', name: '', hash: '', imageUrl: '' },
```
Comment (AlikRakhmonov) [inline frontend/src/scenes/Workspace/reducers/reducer.ts:19]: incorrect, it should be IWorkspace

File: frontend/src/scenes/Workspace/containers/WorkspaceToolbar/index.tsx (hunk 4 above)
Comment (GarryFirst-cripto) [reply frontend/src/scenes/Workspace/containers/WorkspaceToolbar/index.tsx:18]: Done

File: frontend/src/scenes/Workspace/reducers/reducer.ts (hunk 5 above)
Comment (GarryFirst-cripto) [reply frontend/src/scenes/Workspace/reducers/reducer.ts:9]: Done

File: frontend/src/scenes/Workspace/reducers/reducer.ts (hunk 6 above)
Comment (GarryFirst-cripto) [reply frontend/src/scenes/Workspace/reducers/reducer.ts:17]: Done

File: frontend/src/common/models/user/IUser.ts (hunk 1 above)
Comment (AlikRakhmonov) [reply frontend/src/common/models/user/IUser.ts]: waiting for changes

File: frontend/src/scenes/Workspace/reducers/reducer.ts (hunk 5 above)
Comment (AlikRakhmonov) [reply frontend/src/scenes/Workspace/reducers/reducer.ts:9]: it's still string

File: frontend/src/scenes/Workspace/reducers/reducer.ts (hunk 6 above)
Comment (AlikRakhmonov) [reply frontend/src/scenes/Workspace/reducers/reducer.ts:17]: I don't see changes

**Pull request merged**